File: ccr-ui/backend/src/handlers/iflow.rs
// iFlow CLI API 处理器（完整 Stub 实现）

use axum::{
    extract::Path,
    http::StatusCode,
    response::IntoResponse,
    Json,
};
use serde_json::json;

// ============ MCP 服务器管理（Stub 实现）============

/// GET /api/iflow/mcp - 列出所有 MCP 服务器
pub async fn list_iflow_mcp_servers() -> impl IntoResponse {
    (
        StatusCode::OK,
        Json(json!({
            "success": true,
            "data": [],
            "message": "iFlow MCP 服务器功能待实现"
        })),
    )
}

/// POST /api/iflow/mcp - 添加 MCP 服务器
pub async fn add_iflow_mcp_server(Json(_request): Json<serde_json::Value>) -> impl IntoResponse {
    (
        StatusCode::NOT_IMPLEMENTED,
        Json(json!({
            "success": false,
            "data": null,
            "message": "iFlow MCP 服务器添加功能待实现"
        })),
    )
}

/// PUT /api/iflow/mcp/:name - 更新 MCP 服务器
pub async fn update_iflow_mcp_server(
    Path(_name): Path<String>,
    Json(_request): Json<serde_json::Value>,
) -> impl IntoResponse {
    (
        StatusCode::NOT_IMPLEMENTED,
        Json(json!({
            "success": false,
            "data": null,
            "message": "iFlow MCP 服务器更新功能待实现"
        })),
    )
}

/// DELETE /api/iflow/mcp/:name - 删除 MCP 服务器
pub async fn delete_iflow_mcp_server(Path(_name): Path<String>) -> impl IntoResponse {
    (
        StatusCode::NOT_IMPLEMENTED,
        Json(json!({
            "success": false,
            "data": null,
            "message": "iFlow MCP 服务器删除功能待实现"
        })),
    )
}

// ============ Agents 管理（Stub 实现）============

/// GET /api/iflow/agents - 列出所有 Agents
pub async fn list_iflow_agents() -> impl IntoResponse {
    (
        StatusCode::OK,
        Json(json!({
            "success": true,
            "data": {
                "agents": [],
                "folders": []
            },
            "message": "iFlow Agents 功能待实现"
        })),
    )
}

/// POST /api/iflow/agents - 添加 Agent
pub async fn add_iflow_agent(Json(_request): Json<serde_json::Value>) -> impl IntoResponse {
    (
        StatusCode::NOT_IMPLEMENTED,
        Json(json!({
            "success": false,
            "data": null,
            "message": "iFlow Agents 添加功能待实现"
        })),
    )
}

/// PUT /api/iflow/agents/:name - 更新 Agent
pub async fn update_iflow_agent(
    Path(_name): Path<String>,
    Json(_request): Json<serde_json::Value>,
) -> impl IntoResponse {
    (
        StatusCode::NOT_IMPLEMENTED,
        Json(json!({
            "success": false,
            "data": null,
            "message": "iFlow Agents 更新功能待实现"
        })),
    )
}

/// DELETE /api/iflow/agents/:name - 删除 Agent
pub async fn delete_iflow_agent(Path(_name): Path<String>) -> impl IntoResponse {
    (
        StatusCode::NOT_IMPLEMENTED,
        Json(json!({
            "success": false,
            "data": null,
            "message": "iFlow Agents 删除功能待实现"
        })),
    )
}

/// PUT /api/iflow/agents/:name/toggle - 切换 Agent 启用状态
pub async fn toggle_iflow_agent(Path(_name): Path<String>) -> impl IntoResponse {
    (
        StatusCode::NOT_IMPLEMENTED,
        Json(json!({
            "success": false,
            "data": null,
            "message": "iFlow Agents 切换功能待实现"
        })),
    )
}

// ============ Slash Commands 管理（Stub 实现）============

/// GET /api/iflow/slash-commands - 列出所有 Slash Commands
pub async fn list_iflow_slash_commands() -> impl IntoResponse {
    (
        StatusCode::OK,
        Json(json!({
            "success": true,
            "data": {
                "commands": [],
                "folders": []
            },
            "message": "iFlow Slash Commands 功能待实现"
        })),
    )
}

/// POST /api/iflow/slash-commands - 添加 Slash Command
pub async fn add_iflow_slash_command(Json(_request): Json<serde_json::Value>) -> impl IntoResponse {
    (
        StatusCode::NOT_IMPLEMENTED,
        Json(json!({
            "success": false,
            "data": null,
            "message": "iFlow Slash Commands 添加功能待实现"
        })),
    )
}

/// PUT /api/iflow/slash-commands/:name - 更新 Slash Command
pub async fn update_iflow_slash_command(
    Path(_name): Path<String>,
    Json(_request): Json<serde_json::Value>,
) -> impl IntoResponse {
    (
        StatusCode::NOT_IMPLEMENTED,
        Json(json!({
            "success": false,
            "data": null,
            "message": "iFlow Slash Commands 更新功能待实现"
        })),
    )
}

/// DELETE /api/iflow/slash-commands/:name - 删除 Slash Command
pub async fn delete_iflow_slash_command(Path(_name): Path<String>) -> impl IntoResponse {
    (
        StatusCode::NOT_IMPLEMENTED,
        Json(json!({
            "success": false,
            "data": null,
            "message": "iFlow Slash Commands 删除功能待实现"
        })),
    )
}

/// PUT /api/iflow/slash-commands/:name/toggle - 切换 Slash Command 启用状态
pub async fn toggle_iflow_slash_command(Path(_name): Path<String>) -> impl IntoResponse {
    (
        StatusCode::NOT_IMPLEMENTED,
        Json(json!({
            "success": false,
            "data": null,
            "message": "iFlow Slash Commands 切换功能待实现"
        })),
    )
}

// ============ Plugins 管理（Stub 实现）============

/// GET /api/iflow/plugins - 列出所有 Plugins
pub async fn list_iflow_plugins() -> impl IntoResponse {
    (
        StatusCode::OK,
        Json(json!({
            "success": true,
            "data": [],
            "message": "iFlow Plugins 功能待实现"
        })),
    )
}

/// POST /api/iflow/plugins - 添加 Plugin
pub async fn add_iflow_plugin(Json(_request): Json<serde_json::Value>) -> impl IntoResponse {
    (
        StatusCode::NOT_IMPLEMENTED,
        Json(json!({
            "success": false,
            "data": null,
            "message": "iFlow Plugins 添加功能待实现"
        })),
    )
}

/// PUT /api/iflow/plugins/:id - 更新 Plugin
pub async fn update_iflow_plugin(
    Path(_id): Path<String>,
    Json(_request): Json<serde_json::Value>,
) -> impl IntoResponse {
    (
        StatusCode::NOT_IMPLEMENTED,
        Json(json!({
            "success": false,
            "data": null,
            "message": "iFlow Plugins 更新功能待实现"
        })),
    )
}

/// DELETE /api/iflow/plugins/:id - 删除 Plugin
pub async fn delete_iflow_plugin(Path(_id): Path<String>) -> impl IntoResponse {
    (
        StatusCode::NOT_IMPLEMENTED,
        Json(json!({
            "success": false,
            "data": null,
            "message": "iFlow Plugins 删除功能待实现"
        })),
    )
}

/// PUT /api/iflow/plugins/:id/toggle - 切换 Plugin 启用状态
pub async fn toggle_iflow_plugin(Path(_id): Path<String>) -> impl IntoResponse {
    (
        StatusCode::NOT_IMPLEMENTED,
        Json(json!({
            "success": false,
            "data": null,
            "message": "iFlow Plugins 切换功能待实现"
        })),
    )
}

File: ccr-ui/frontend/src/api/modules/iflow.ts
/**
 * iFlow Platform API Module
 *
 * iFlow MCP、Agents/SlashCommands/Plugins 管理
 */

import { api } from '../core'
import type {
    ApiResponse,
    Plugin,
    PlatformAgentsResponse,
    PlatformAgentRequest,
    PlatformSlashCommandsResponse,
    PlatformSlashCommandRequest,
    PlatformPluginRequest,
    PlatformMcpServerRequest,
} from '@/types'

// ═══════════════════════════════════════════════════════════
// 🔌 iFlow MCP Server Management
// ═══════════════════════════════════════════════════════════

export const listIflowMcpServers = async (): Promise<PlatformMcpServerRequest[]> => {
    const response = await api.get<ApiResponse<PlatformMcpServerRequest[]>>('/iflow/mcp')
    return response.data.data ?? (response.data as any)
}

export const addIflowMcpServer = async (_request: PlatformMcpServerRequest): Promise<string> => {
    const response = await api.post<ApiResponse<string>>('/iflow/mcp', _request)
    return response.data.message || 'OK'
}

export const updateIflowMcpServer = async (_name: string, _request: PlatformMcpServerRequest): Promise<string> => {
    const response = await api.put<ApiResponse<string>>(`/iflow/mcp/${encodeURIComponent(_name)}`, _request)
    return response.data.message || 'OK'
}

export const deleteIflowMcpServer = async (_name: string): Promise<string> => {
    const response = await api.delete<ApiResponse<string>>(`/iflow/mcp/${encodeURIComponent(_name)}`)
    return response.data.message || 'OK'
}

// ═══════════════════════════════════════════════════════════
// 🤖 iFlow Agents
// ═══════════════════════════════════════════════════════════

export const listIflowAgents = async (): Promise<PlatformAgentsResponse> => {
    const response = await api.get<ApiResponse<PlatformAgentsResponse>>('/iflow/agents')
    return response.data.data ?? (response.data as any)
}

export const addIflowAgent = async (_request: PlatformAgentRequest): Promise<string> => {
    throw new Error('Backend API not implemented yet')
}

export const updateIflowAgent = async (_name: string, _request: PlatformAgentRequest): Promise<string> => {
    throw new Error('Backend API not implemented yet')
}

export const deleteIflowAgent = async (_name: string): Promise<string> => {
    throw new Error('Backend API not implemented yet')
}

export const toggleIflowAgent = async (_name: string): Promise<string> => {
    const response = await api.put<ApiResponse<string>>(`/iflow/agents/${encodeURIComponent(_name)}/toggle`)
    return response.data.message || 'OK'
}

// ═══════════════════════════════════════════════════════════
// ⚡ iFlow Slash Commands
// ═══════════════════════════════════════════════════════════

export const listIflowSlashCommands = async (): Promise<PlatformSlashCommandsResponse> => {
    const response = await api.get<ApiResponse<PlatformSlashCommandsResponse>>('/iflow/slash-commands')
    return response.data.data ?? (response.data as any)
}

export const addIflowSlashCommand = async (_request: PlatformSlashCommandRequest): Promise<string> => {
    const response = await api.post<ApiResponse<string>>('/iflow/slash-commands', _request)
    return response.data.message || 'OK'
}

export const updateIflowSlashCommand = async (_name: string, _request: PlatformSlashCommandRequest): Promise<string> => {
    const response = await api.put<ApiResponse<string>>(`/iflow/slash-commands/${encodeURIComponent(_name)}`, _request)
    return response.data.message || 'OK'
}

export const deleteIflowSlashCommand = async (_name: string): Promise<string> => {
    const response = await api.delete<ApiResponse<string>>(`/iflow/slash-commands/${encodeURIComponent(_name)}`)
    return response.data.message || 'OK'
}

export const toggleIflowSlashCommand = async (_name: string): Promise<string> => {
    const response = await api.put<ApiResponse<string>>(`/iflow/slash-commands/${encodeURIComponent(_name)}/toggle`)
    return response.data.message || 'OK'
}

// ═══════════════════════════════════════════════════════════
// 🔌 iFlow Plugins
// ═══════════════════════════════════════════════════════════

export const listIflowPlugins = async (): Promise<Plugin[]> => {
    const response = await api.get<ApiResponse<Plugin[]>>('/iflow/plugins')
    return response.data.data ?? (response.data as any)
}

export const addIflowPlugin = async (_request: PlatformPluginRequest): Promise<string> => {
    throw new Error('Backend API not implemented yet')
}

export const updateIflowPlugin = async (_id: string, _request: PlatformPluginRequest): Promise<string> => {
    throw new Error('Backend API not implemented yet')
}

export const deleteIflowPlugin = async (_id: string): Promise<string> => {
    throw new Error('Backend API not implemented yet')
}

export const toggleIflowPlugin = async (_id: string): Promise<string> => {
    throw new Error('Backend API not implemented yet')
}
